documentation/docs/developer_guide/contribution_flowchart.md

```diff
@@ -0,0 +1,162 @@
+---
+hide:
+  - toc
+---
+
+# Flowchart to contribute to CABLE's documentation
+
+```mermaid
+   %% Create a graph for the legend of the main graph
+   flowchart LR
+      uniq[action done <br> only once ever <br> at the start <br> to get the code locally]:::uniq;
+      sevterm[action done <br> several times per issue <br> in a terminal/text editor]:::term;
+      sevgit[action done <br> several times per issue <br> on GitHub]:::github;
+      onceterm([action done <br> once per issue <br> in a terminal/text editor]):::term;
+      oncegit([action done <br> once per issue <br> on GitHub]):::github;
+      question[\question with multiple outcomes/]:::question;
+
+      uniq --- sevterm --- sevgit --- question;
+      uniq --- onceterm --- oncegit --- question;
+
+      classDef default fill:#FFFDE7, stroke:#FFF59D;
+      classDef uniq fill:#D81B60, stroke:#880E4F, color:#FFFFFF;
+      classDef github fill:#388E3C,stroke:#1B5E20, color:#FFFFFF;
+      classDef term fill:#F44336, stroke:#B71C1C, color:#FFFFFF;
+      classDef question fill:#6D4C41, stroke:#3E2723, color:#FFFFFF; 
+;
+```
+
+```mermaid
+   flowchart TB
+
+      %% Define all the nodes first
+      clone[Clone repository]:::uniq ;
+      issue([Open an issue <br> to explain your work]):::github;
+      branch([Create a branch]):::github;
```

Comment:
Why create a branch on GitHub and then fetch? Why not just create locally? It also removes the `fetch` step, making it simpler.

Reply:
But then you have to use `push -u origin <branchname>` instead of `push`. And then people choose their branch name and you loose uniformity of the branch name. But maybe that's not a big deal if people create PR.

Reply:
I am drawing a flowchart and will send shortly...I think it is best to start always by creating the branch on GitHub and THEN do a clone or pull. In this case you automatically get a local branch that tracks upstream. No need for fetch or creating explicitly a local branch.

Reply:
I have literally *never* done this. I'm not saying this can't be the workflow for this repo, but from my experience that is unusual.

I can see that it is nice to a link for GitHub create a branch for you and have a uniform branch name.

> But then you have to use `push -u origin <branchname>` instead of `push`.

Depending on your git settings you can still just `git push` and it'll default to your local branch name.

Reply:
You mean this setting @aidanheerdegen : `git config --global push.autoSetupRemote true` I had never heard of it. I wonder what happens when you have several remotes... But I can look into that myself.

Reply:
@rkutteh Why are you drawing a flowchart? You don't like mine?

Reply:
On the contrary, your chart inspired me to draw one for myself with slight modifications. I will also use it for the CMS work. I will share it shortly but I think yours is already very good that you probably won't benefit from mine.

Reply:
Another advantage of telling people to create the branch from the Issue screen is that it gets people to start by putting an issue in. Otherwise, it's very easy to start something on your own machine and then only push at the very end.

Reply:
> Do you reckon it is worth adding this setting to the Git tutorial we have so people set up their git this way "by default"?

Seems mostly good and free of harm. It is how one would expect it to work IMO. It is a pretty recent change, so one downside is they might have to take care to use a newish version of `git`

https://pawelgrzybek.com/auto-setup-remote-branch-and-never-again-see-an-error-about-the-missing-upstream/

Reply:
> Another advantage of telling people to create the branch from the Issue screen is that it gets people to start by putting an issue in. Otherwise, it's very easy to start something on your own machine and then only push at the very end.

I can see the advantage of being quite prescriptive, but it risks hiding the point of what you're trying to achieve behind a bunch of steps that it's not clear which are the most important, or why exactly it is being done that way.

If someone was, say, trying to use VS Code with the GitHub extension it wouldn't make sense to follow this advice exactly, but the outcome would be no less useful.

So another way to do it might be "this is what we're trying to achieve" flowchart, and "this is *how* you could do that with one specific method".

Reply:
@ccarouge @aidanheerdegen My pocket (pirated) version of your workflow Claire, with a slight change at the top. No doubt I have omitted a git flag here or there. I will make corrections as I follow it in practice ...Thanks!
[Git_Workflow.pdf](https://github.com/CABLE-LSM/CABLE/files/10137967/Git_Workflow.pdf)